src/planners/bidirectional.jl

```diff
@@ -0,0 +1,130 @@
+export BiDirectionalPlanner, BiGreedyPlanner, BiAStarPlanner
+
+"BiDirectionalPlanner best-first search planner."
+mutable struct BiDirectionalPlanner{F,B} <: Planner
+    forward::F
+    backward::B
+end
+
+BiDirectionalPlanner(;kwargs...) = BiDirectionalPlanner(
+    	ForwardPlanner(; kwargs...),
+    	BackwardPlanner(; kwargs...)
+    )
+
+"Bidirectional  Greedy best-first search, with cycle checking."
+BiGreedyPlanner(heuristic::Heuristic; kwargs...) =
+    BiDirectionalPlanner(
+    	ForwardPlanner(;heuristic=heuristic, g_mult=0, kwargs...),
+    	BackwardPlanner(;heuristic=heuristic, g_mult=0, kwargs...)
+    )
+
+
+"Bidirectional A* search."
+BiAStarPlanner(heuristic::Heuristic; kwargs...) =
+    BiDirectionalPlanner(
+    	ForwardPlanner(;heuristic=heuristic, kwargs...),
+    	BackwardPlanner(;heuristic=heuristic, kwargs...)
+    )
```

Comment:
For both of these constructors, I think the forward heuristic and backward heuristic should actually be specified separately, rather than shared across both planners. This is because some heuristics, like `HAdd`, only make sense when used with a `ForwardPlanner`, whereas `HAddR` is only meant to be used with a `BackwardPlanner`.

Maybe replace with.

```julia
BiAStarPlanner(f_heuristic::Heuristic, b_heuristic::Heuristic; kwargs...) =
    BiDirectionalPlanner(
    	ForwardPlanner(;heuristic=f_heuristic, kwargs...),
    	BackwardPlanner(;heuristic=b_heuristic, kwargs...)
    )
```

I would also adjust all of the other constructors similarly.